src/models/utils.py

```diff
@@ -44,7 +43,7 @@ def _chunk_ndarray(x: Tuple[np.ndarray, ...], y: np.ndarray,
                    num_sections: int,
                    shuffle: bool) -> Iterable[Tuple[Tuple[np.ndarray, ...], np.ndarray]]:
 
-    split_x = [np.array_split(x_section, num_sections) for x_section in x]
+    split_x = [np.array_split(x_section, num_sections) for x_section in x if x_section is not None]
```

Comment:
chunk will return a different sized Tuple than it receives, without telling the user that it does that

Reply:
This should be fixed since the indexing will get screwed up depending on what arguments are passed to the model

Reply:
Sorry I don't understand this one? What do I need to change here? To tell the user or to have some assert statements to make sure everything is still in the same dimensions?

Reply:
or to change the chunking itself?

Reply:
Fixed!